x-pack/metricbeat/module/panos/client.go

```diff
@@ -0,0 +1,45 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package panos
+
+import (
+	"flag"
+
+	"github.com/PaloAltoNetworks/pango"
+)
+
+// PanosClient interface with an Op function
+type PanosClient interface {
+	Op(req interface{}, vsys string, extras interface{}, ans interface{}) ([]byte, error)
+}
+
+type PanosFirewallClient struct {
+	pango.Firewall
+}
+
+type PanosTestClient struct {
+}
+
+// Op is a mock function for testing that returns sample XML output based on the initial req parameter
+// XML output is stored in the testdata directory, one file per query string
+func (c *PanosTestClient) Op(req interface{}, vsys string, extras, ans interface{}) ([]byte, error) {
+	return nil, nil
+}
+
+func GetPanosClient(config *Config) (PanosClient, error) {
+	// If running tests, return a test client
+	if flag.Lookup("test.v") != nil {
+		return &PanosTestClient{}, nil
+	}
+
+	firewall := pango.Firewall{Client: pango.Client{Hostname: config.HostIp, ApiKey: config.ApiKey}}
```

Comment:
From firewall.Initialize's doc, I notice that if necessary args for params are not perovided then the following default is taken:

"""
Protocol: https
Port: (unspecified)
Timeout: 10
Logging: LogAction | LogUid
"""

As here, we are just providing Hostname and ApiKey with the client; are we good with the defaults? For example, always "https"; then shouldn't we take certificates from the config too? Also, the significance of a port in client?

Reply:
I don't see anywhere in the pango client where it is handling certificates specifically. The only configuration parameter related to certs is
`VerifyCertificate bool            `json:"verify_certificate"``
and this is only used to configure the https client:
`	// Setup the https client.
	if c.Transport == nil {
		c.Transport = &http.Transport{
			Proxy: http.ProxyFromEnvironment,
			TLSClientConfig: &tls.Config{
				InsecureSkipVerify: !c.VerifyCertificate,
			},
		}
	}`
	
	Setting that InsecureSkipVerify to false only forces verification of the server certificates.

Reply:
I suppose it makes sense to support setting an alternative port, though. I have no idea of what standard practice is for Palo Alto firewall admins....